designs/lily/src/back.mjs

```diff
@@ -0,0 +1,567 @@
+      })
+      macro('hd', {
+        from: points.floorIn,
+        to: points.floor,
+        y: points.floorIn.y - 15,
+      })
+      macro('hd', {
+        from: points.floor,
+        to: points.floorOut,
+        y: points.floorIn.y - 15,
+      })
+      macro('vd', {
+        from: points.floorOut,
+        to: points.styleWaistOutLily,
+        x:
+          (points.seatOut.x > points.styleWaistOutLily.x ? points.seatOut.x : points.styleWaistOutLily.x) +
+          sa +
+          15,
+      })
+      macro('vd', {
+        from: points.floorIn,
+        to: points.fork,
+        x: points.fork.x - sa - 15,
+      })
+      macro('vd', {
+        from: points.fork,
+        to: points.styleWaistInLily,
+        x: points.fork.x - sa - 15,
+      })
+      macro('vd', {
+        from: points.floorIn,
+        to: points.styleWaistInLily,
+        x: points.fork.x - sa - 30,
+      })
+      macro('vd', {
+        from: points.crossSeamCurveStart,
+        to: points.styleWaistInLily,
+        x: points.crossSeamCurveStart.x - sa - 15,
+      })
+      macro('hd', {
+        from: points.styleWaistInLily,
+        to: points.grainlineTop,
+        y: points.styleWaistInLily.y - sa - 15,
+      })
+      macro('hd', {
+        from: points.crossSeamCurveStart,
+        to: points.grainlineTop,
+        y: points.styleWaistInLily.y - sa - 30,
+      })
+      macro('hd', {
+        from: points.crossSeamCurveMax,
+        to: points.grainlineTop,
+        y: points.styleWaistInLily.y - sa - 45,
+      })
+      macro('hd', {
+        from: points.fork,
+        to: points.grainlineTop,
+        y: points.styleWaistInLily.y - sa - 60,
+      })
+      macro('hd', {
+        from: points.grainlineTop,
+        to: points.styleWaistOutLily,
+        y: points.styleWaistInLily.y - sa - 15,
+      })
+      if (points.seatOut.x > points.styleWaistOutLily.x) {
+        macro('hd', {
+          from: points.grainlineTop,
+          to: points.seatOut,
+          y: points.styleWaistInLily.y - sa - 30,
+        })
+      }
+    }
+  }
+
+  return part
+}
+
+export const back = {
+  name: 'lily.back',
+  from: titanBack,
+  //after: titanFront,
+  measurements: [
+    'ankle',
+    'heel', // secondary measurement, used instead of ankle
+  ],
+  options: {
+    fitGuides: {bool: true, menu: undefined},
+    fitKnee: {bool: true, menu: undefined},
+    legBalance: 0.5, // between back and front parts
+    waistBalance: 0.5,
+    waistHeight: { ...titanBack.options.waistHeight, pct: 50 }, // halfway between waist and hips
+    fabricStretch: { pct: 40, min: 0, max: 50, menu: 'fit' },
+    waistEase: {pct: -4,  menu: undefined}, // -fabricStretch/10,
+    seatEase: {pct: -4, menu: undefined}, // -fabricStretch/10,
+    kneeEase: {pct: -4, menu: undefined}, // -fabricStretch/10,
+    test: {pct: ({ options }) => (options.fabricStretch/2)},
+    lengthBonus: 0,
+    lengthReduction: { pct: 0, min: 0, max: 100, menu: 'style'},
```

Comment:
100% seems like a large maximum value for lengthReduction. 

I note that the pattern crashes with an error if Doll cisFemale 10% is used and lengthReduction is 42.4-43.3%. It crashes again with lengthReduction >= 71.5%.

With cisFemale size 28, it crashes with lengthReduction 45.3-45.6% and 73.6%.

Reply:
This seems to occur consistently when the leggings are cut off _at_ the knee (rather than above or below). It's a bit tricky to fix, so it'll have to wait until a newer version.

Reply:
As for the lengthReduction maximum value: it actually caps when the leggings end at the fork, with the following log message: "length reduction capped; cutting off at fork".

The percentage this ends up being depends on the measurements, so I just let it go all the way to 100%.